internal/metrics/benchmark_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package benchmark_test
+
+import (
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+
+	"github.com/jaegertracing/jaeger/internal/metrics/otelmetrics"
+	prom "github.com/jaegertracing/jaeger/internal/metrics/prometheus"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+)
+
+func benchmarkCounter(b *testing.B, factory metrics.Factory) {
+	counter := factory.Counter(metrics.Options{
+		Name: "test_counter",
+		Tags: map[string]string{"tag1": "value1"},
+	})
+
+	for i := 0; i < b.N; i++ {
+		counter.Inc(1)
+	}
+}
+
+func BenchmarkPrometheusCounter(b *testing.B) {
+	reg := prometheus.NewRegistry()
+	factory := prom.New(prom.WithRegisterer(reg))
+	benchmarkCounter(b, factory)
+}
+
+func BenchmarkOTELCounter(b *testing.B) {
+	res := resource.NewWithAttributes(
+		resource.Default().SchemaURL(),
+	)
+	meterProvider := metric.NewMeterProvider(metric.WithResource(res))
```

Comment:
you are not defining an exporter. For comparison it would be good to use prometheus exporter, similar to the other benchmark.

Reply:
Example: https://github.com/open-telemetry/opentelemetry-go/blob/7e0813445b42443b911f02a1156292120cec98bc/exporters/prometheus/exporter_test.go#L428